tutorials/plugins/gdnative/files/cpp_example/SConstruct

```diff
@@ -1,48 +1,92 @@
 #!python
-import os
+import os, subprocess
 
-target = ARGUMENTS.get("target", "debug")
-platform = ARGUMENTS.get("platform", "windows")
-bits = ARGUMENTS.get("bits", 64)
+opts = Variables([], ARGUMENTS)
 
-final_lib_path = 'demo/bin/'
+# Gets the standard flags CC, CCX, etc.
+env = DefaultEnvironment()
 
-# This makes sure to keep the session environment variables on windows,
-# that way you can run scons in a vs 2017 prompt and it will find all the required tools
-env = Environment()
-if platform == "windows":
-    env = Environment(ENV = os.environ)
+# Define our options
+opts.Add(EnumVariable('target', "Compilation target", 'debug', ['d', 'debug', 'r', 'release']))
+opts.Add(EnumVariable('platform', "Compilation platform", 'windows', ['windows', 'x11', 'linux', 'osx']))
+opts.Add(EnumVariable('p', "Compilation target, alias for 'platform'", 'not set', ['windows', 'x11', 'linux', 'osx', 'not set']))
+opts.Add(BoolVariable('use_llvm', "Use the LLVM / Clang compiler", 'no'))
+opts.Add(PathVariable('target_path', 'The path where the lib is installed.', 'demo/bin/'))
+opts.Add(PathVariable('target_name', 'The library name.', 'libgdexample', PathVariable.PathAccept))
 
-def add_sources(sources, directory):
-    for file in os.listdir(directory):
-        if file.endswith('.cpp'):
-            sources.append(directory + '/' + file)
+# Local dependency paths, adapt them to your setup
+godot_headers_path = "godot-cpp/godot_headers/"
+cpp_bindings_path = "godot-cpp/"
+cpp_library = "libgodot-cpp"
 
-if platform == "osx":
-    env.Append(CCFLAGS = ['-g','-O3', '-arch', 'x86_64', '-std=c++14'])
-    env.Append(LINKFLAGS = ['-arch', 'x86_64'])
+# only support 64 at this time..
+bits = 64
 
-    final_lib_path = final_lib_path + 'osx/'
+# Updates the environment with the option variables.
+opts.Update(env)
 
-elif platform == "linux":
-    env.Append(CCFLAGS = ['-fPIC', '-g','-O3', '-std=c++14'])
+# Process some arguments
+if env['use_llvm']:
+    env['CXX'] = 'clang++'
```

Comment:
Should also at least set `env['CC']` to `clang`, in case people also build a C library together with their C++ GDNative code.

Reply:
No problem